apmpackage/apm/data_stream/internal_metrics/elasticsearch/ingest_pipeline/default.yml

```diff
@@ -13,10 +13,12 @@ processors:
       name: process_ppid
   - pipeline:
       name: client_geoip
-  - remove:
-      field: _metric_descriptions
-      ignore_missing: true
   - remove:
       # Removed in 8.6.0
       field: timeseries.instance
       ignore_missing: true
+  - pipeline:
+      name: set_metrics
+  - remove:
+      field: _dynamic_templates
+      ignore_missing: true
```

Comment:
Why is the remove pipeline necessary here? Previously, the `set_metrics` script wasn't called here, but now that it is, it removes the `_metric_descriptions` and `samples` anyways. 
I also generally don't follow why `_dynamic_templates` should now be removed?

Reply:
Note that we're removing `_dynamic_templates` here, not `_metric_descriptions. `_dynamic_templates` is related to dynamic mapping -- it's set by the `set_metrics` pipeline to tell Elasticsearch how to dynamically map a field.

We only rely on dynamic mapping for the app_metrics data stream. We were previously relying on metrics being set at the top level of the document, and we removed `_metric_descriptions`, which was sent by APM Server. Now we're sending `metricset.samples` instead, which maps metrics to the top level and sets `_dynamic_templates`. We want the first bit, but not the second.